src/mime_app.rs

```diff
@@ -10,16 +10,20 @@ use std::{
     cmp::Ordering, collections::HashMap, env, path::PathBuf, process, sync::Mutex, time::Instant,
 };
 
-pub fn exec_to_command(exec: &str, path_opt: Option<PathBuf>) -> Option<process::Command> {
+pub fn exec_to_command(exec: &str, path_opt: Vec<Option<PathBuf>>) -> Option<process::Command> {
     let args_vec: Vec<String> = shlex::split(exec)?;
     let mut args = args_vec.iter();
     let mut command = process::Command::new(args.next()?);
     for arg in args {
         if arg.starts_with('%') {
             match arg.as_str() {
                 "%f" | "%F" | "%u" | "%U" => {
```

Comment:
If a program provides %f or %u it means only a single path or URL. %F or %U can accept a list.

Reply:
Oh, did not know. To be honest I was reading about it today, because testing something but I did not consider it here :sweat: Should I split this match pattern and in this case provide only first argument from vector and full list in other case?